packages/graph-editor/src/components/controls/array.tsx

```diff
@@ -0,0 +1,178 @@
+import { observer } from 'mobx-react-lite';
+import { JSONTree } from 'react-json-tree';
+import React from 'react';
+import { IField } from './interface';
+import { toJS } from 'mobx';
+import { Button, IconButton, Select, Stack, TextInput } from '@tokens-studio/ui';
+import { Minus, Plus } from 'iconoir-react';
+import { ColorPickerPopover } from '../colorPicker';
+import { ANY, AllSchemas, COLOR, Input, NUMBER, STRING } from '@tokens-studio/graph-engine';
+import { delayedUpdateSelector } from '@/redux/selectors';
+import { useSelector } from 'react-redux';
+
+const inputItemTypes = [STRING, NUMBER, COLOR];
+
+export const ArrayField = observer(({ port }: IField) => {
```

Comment:
Looks like we're redefining how these types work internally. I think the better approach would be that we don't pass the raw port directly to an input type and rather pass the value  and have a managed system to handle the updates. 

Then we can support an array of any arbitrary type if there is a control registered for it

Reply:
This is exactly how all other input types are handled. What do you mean by "we're redefining how these types work internally"?

Reply:
No, this is the opposite of how the other controls work. Controls work by using a selector to choose the appropriate control. Thus I can create and register a controls for a specific type foo and all that needs to happen is that a foo value is present and the port panel will show it automatically.

This overrides the array control to then specifically handle certain types. This means its not scalable if a new type foo is present and someone would need to both create a specific array foo control as well as register it with higher priority than this array control.

If we rather abstract the array control to then only handle the array portion and render any arbitrary control inside of it, it will be much more scalable, and only if there is a specific use case to override the default array handling would someone need to create a new control

Reply:
![image](https://github.com/tokens-studio/graph-engine/assets/19837930/e86273c9-df62-4986-aeb6-3eda19b9b7c6)

All controls receive the port and then use the value to show the inputs or whatever. Do you mean to implement the array as you did vor variadic controls? I can do that, of course :D

Reply:
So I think having the array type then recursively use the selector logic to then choose the type inside of itself could be optimal. Then we should be able to do things like have an array of array of numbers for example and it should continue to work regardless of the depth of array nesting if it always chooses the next appropriate control inside of itself

Reply:
I understand what you mean but it wouldn't be doable using our current controls (string, color ...) because within the control the change is being pushed on the received por. Our port will be the array, and the change should be performed on the array port, not a type inside the array. I will create new controls for `ArrayColor`, `ArrayNumber`, `ArrayString` and others in the future, but I am not able to use our current controls for color, number and array.